decidim/decidim
# Embedded jitsi meetings

#### :tophat: What? Why?
This new feature lets Jitsi Meet rooms to be automatically generated and embedded in the meeting page when an online meeting is created by an admin.

The Jitsi server and the Jisi API used for this feature can be configured in `secrets.yml` and in the `decidim.rb` initializer.

It also adds a system to log users attendance to the videoconference, and view it from the admin for meetings component

#### :pushpin: Related Issues
- Related to MetaDecidim proposal [\#15596](https://meta.decidim.org/processes/roadmap/f/122/proposals/15596)
- Inspired by [alabs/decidim-module-jitsi-meetings](https://github.com/alabs/decidim-module-jitsi-meetings) and [this snippet](https://gist.github.com/andreslucena/9c293f4ba916a3e7da164720da0af21b) by @andreslucena

#### Testing
- Create a meeting and check the "Embedded videoconference" option
- Visit the meeting within its start_time and end_time
- Click on the "Join videoconference" button
- Leave the videoconference
- Go to the admin panel to manage the meetings component and click on the "pulse" icon
- View the attendance logs for the videoconference 

#### :clipboard: Checklist
- [x] :heavy_check_mark: **DO** add seeds.

- [ ] Add more system specs
- [ ] Add button for easy videoconference fullscreen
- [x] Add example values for `JITSI_DOMAIN` and `JITSI_API_URL` 

### :camera: Screenshots

#### User

![Join videoconference](https://user-images.githubusercontent.com/8806781/102893268-978c4180-4461-11eb-967f-03337bd7776d.png)

![Videoconference](https://user-images.githubusercontent.com/8806781/102893347-bbe81e00-4461-11eb-9bce-230a3a706067.png)

![Videoconference](https://user-images.githubusercontent.com/8806781/102893410-d0c4b180-4461-11eb-80c1-badaf63e8133.png)

#### Admin

![Admin - Edit meeting](https://user-images.githubusercontent.com/8806781/102903180-4172ca80-4470-11eb-9210-97c9746d0573.gif)

![Admin - See Attendance Logs](https://user-images.githubusercontent.com/8806781/102893067-35cbd780-4461-11eb-9e31-2ef770de8c11.gif)

#### Improve seeded meetings avoiding duplicated names

![Admin - Seeds](https://user-images.githubusercontent.com/8806781/102906137-7ed95700-4474-11eb-8e88-2cc8618ef593.png)



:hearts: Thank you!
